mne/annotations.py

```diff
@@ -647,6 +647,216 @@ def rename(self, mapping, verbose=None):
         return self
 
 
+class EpochAnnotationsMixin:
+    """Mixin class for Annotations in Epochs."""
+
+    @property
+    def annotations(self):  # noqa: D102
+        return self._annotations
+
+    @verbose
+    def set_annotations(self, annotations, on_missing='raise', *,
+                        verbose=None):
+        """Setter for Epoch annotations from Raw.
+
+        This function does not handle offsetting the times based
+        on first_samp or measurement dates, since that is expected
+        to occur in Raw.set_annotations().
+
+        Parameters
+        ----------
+        annotations : instance of mne.Annotations | None
+            Annotations to set.
+        %(on_missing_ch_names)s
+        %(verbose)s
+
+        Returns
+        -------
+        self : instance of Epochs
+            The epochs object with annotations.
+
+        Notes
+        -----
+        Annotation onsets and offsets are stored as time in seconds (not as
+        sample numbers). If the user has an Epochs object that did not support
+        Annotations before, and the Epochs were resampled, then the Annotations
+        may not show up in the correct locations. So care must be taken when
+        doing things like decimating or resampling with Epochs objects saved
+        before Annotations were supported in Epochs.
```

Comment:
> I think we can still keep the doc string warnings and such tho right?

Yes we should keep them but adjust them.
```suggestion
        Annotation onsets and offsets are stored as time in seconds (not as
        sample numbers).
        
        If you have an ``-epo.fif`` file saved to disk created before 1.0,
        annotations can be added correctly only if no decimation or
        resampling was performed. We thus suggest to regenerate your
        :class:`mne.Epochs` from raw and re-save to disk with 1.0+ if you
        want to safely work with Annotations in Epochs.
```